enginetest/queries/queries.go

```diff
@@ -7154,6 +7154,14 @@ With c as (
 		Query:    "SELECT 4294967296;",
 		Expected: []sql.Row{{int64(4294967296)}},
 	},
+	{
+		Query: "select * from mytable where exists (select * from mytable where i = 1);",
```

Comment:
join queries go into `join_op_tests.go` and/or `join_planning_tests.go`

Reply:
I think there's a benefit to having the queries in both places:
- adding a query to `queries.go` will verify that we execute the query correctly
- adding a query to `join_op_tests.go` or `join_planning_tests.go` will verify that we understand how this query is being planned

There was an alias disambiguation bug that was only caught because we executed the query: the bug wasn't manifesting if we only verified that the plan but didn't also execute the query.

Reply:
`join_op_tests.go` will test the join with every join operator, rather than just one. Any join in `queries.go` will be more thoroughly tested in `join_op_tests.go`. The planning tests have a narrower use-case, which is that we want to make sure a specific transformation rule is applied to create a specialized join (but they also optionally test correctness).